### PR TITLE
Edit notes (Fixes #27, #361)

### LINKED ISSUE #27: Allow users to make notes on meanings
I would like to be able to see the notes I have on meanings and create new ones.

### LINKED ISSUE #361: Feature request: improve the user note feature
First of all, I'd like to say that I like the Tsurukame app very much, I use it all the time.

Recently, Tsurukame started to show the user note field in the kanji/vocabulary information screen.  I must say, it was a life saver for me.  I use user notes very heavily, and before that I had to switch to the WaniKani app in the mobile Chrome constantly, which was very time consuming.  So thank you for introducing this feature!

I'd like to request a couple improvements:

1. There are two different note fields in WaniKani, the "Meaning Note" field and the "Reading Note" field.  However, Tsurukame only shows one field instead of two.  What would happen if both these fields are non-empty in WaniKani? 

In any case, please **show these two note fields separately** in Tsurukame as well.  I'd say show them below the corresponding "Meaning Mnemonic" and "Reading Mnemonic" fields, just like they are shown in WaniKani.

2. Please enable **adding and editing** the "Meaning Note" and "Reading Note" fields via the Tsurukame app interface.  This way, this feature will be complete.

Thank you.

### PR DESCRIPTION
**Pull request by UInt2048**
Fixes #27 and closes #361 by adding edit meaning/reading notes buttons.

### CODE REVIEW

File: ios/SubjectDetailsView.swift [hunk 1]
```diff
@@ -160,6 +160,12 @@ class SubjectDetailsView: UITableView, SubjectChipDelegate {
       model.addSection("Meaning Note")
       model.add(notesItem)
     }
+    let hasMeaningNote = studyMaterials?.hasMeaningNote != nil
```
Comment (davidsansome) [inline ios/SubjectDetailsView.swift:163]: hasMeaningNote is a bool - I think testing for nil is gonna give you the wrong result when the bool is false.

I would also check if the string is empty to catch the case when the proto field is set to the empty string.

`let hasMeaningNote = !(studyMaterials.meaningNote ?? "").isEmpty`

File: ios/SubjectDetailsView.swift [hunk 2]
```diff
@@ -183,6 +189,12 @@ class SubjectDetailsView: UITableView, SubjectChipDelegate {
       model.addSection("Reading Note")
       model.add(notesItem)
     }
+    let hasReadingNote = studyMaterials?.hasReadingNote != nil
```
Comment (davidsansome) [inline ios/SubjectDetailsView.swift:192]: Same here

File: ios/SubjectDetailsView.swift [hunk 3]
```diff
@@ -431,6 +443,44 @@ class SubjectDetailsView: UITableView, SubjectChipDelegate {
...
+    editNote(subject: subject, studyMaterials: studyMaterials, isMeaningNote: false)
+  }
+
+  func editNote(subject: TKMSubject, studyMaterials: TKMStudyMaterials?,
+                isMeaningNote: Bool) {
+    var materials = studyMaterials ?? TKMStudyMaterials()
+    if studyMaterials == nil {
+      materials.subjectID = subject.id
+    }
+    let hasNote = isMeaningNote ? materials.hasMeaningNote : materials.hasReadingNote
```
Comment (davidsansome) [inline ios/SubjectDetailsView.swift:460]: Make sure the logic is the same here too

**Review by davidsansome — APPROVED** [review]

File: ios/SubjectDetailsView.swift (hunk 1 above)
Comment (UInt2048) [reply ios/SubjectDetailsView.swift]: Thanks for the tip! I've updated these three checks as requested.

**Pull request merged**